src/main/kotlin/uk/gov/justice/digital/hmpps/prisonerfromnomismigration/incentives/IncentiveMappingService.kt

```diff
@@ -108,9 +115,17 @@ class IncentiveMappingService(@Qualifier("mappingApiWebClient") private val webC
     }
     .block()?.count ?: 0
 
-  fun deleteIncentiveMapping(nomisBookingId: Long, nomisIncentiveSequence: Long, incentiveId: Long) {
-    // TODO - delete mapping for completeness though if we don't it just leaves a dangling mapping
-  }
+  suspend fun deleteIncentiveMapping(incentiveId: Long): Unit? = webClient.delete()
+    .uri(
+      "/mapping/incentives/incentive-id/{incentiveId}",
+      incentiveId,
+    )
+    .retrieve()
+    .bodyToMono(Unit::class.java)
+    .onErrorResume(WebClientResponseException::class.java) {
+      log.error("Unable to deleting mapping for incentiveId $incentiveId but ignoring and allowing dangling record", it)
+      Mono.empty()
```

Comment:
decided to ignore error; rather than writing a ton of retry code when worse case scenario we have a dangling record, the important thing is the actual Incentive has been deleted